code/midtown/mmcity/cullcity.cpp

```diff
@@ -85,3 +98,162 @@ void mmCullCity::Cull()
+
+    IsSnowing = (MMSTATE.Weather == mmWeather::Snow);
+    CityName = arts_strdup(name);
+
+    arts_strcpy(city_path, name);
+    arts_strcat(city_path, "city");
+
+    arts_strcpy(lm_path, name);
+    arts_strcat(lm_path, "lm");
+
+    TEXSHEET.allow_remapping_ = (MMSTATE.TimeOfDay == mmTimeOfDay::Sunset or MMSTATE.TimeOfDay == mmTimeOfDay::Night);
+
+    switch (MMSTATE.Weather)
+    {
+        case mmWeather::Rain: TextureSuffix = "_fall"_xconst; break;
+        case mmWeather::Snow: TextureSuffix = "_win"_xconst; break;
+        default: TextureSuffix = nullptr; break;
+    }
+
+    StaticLog = arts_fopen(arts_formatf<64>("%s_static.csv", name), "w");
+
+    Loader()->BeginTask(LOC_STRING(MM_IDS_LOADING_CITY_LAYOUT), 0.0f);
+
+    DLPTemplate* city_dlp = nullptr;
+    DLPTemplate* lm_dlp = nullptr;
+
+    if (DevelopmentMode)
+    {
+        city_dlp = GetDLPTemplate(city_path);
+        if (!city_dlp)
+            Quitf("Unable to load city '%s'", name);
+
+        lm_dlp = GetDLPTemplate(lm_path);
+    }
+
+    BeginMemStat("mmCullCity::Init");
+    InitProblems();
+
+    Camera = camera;
+    asNode::AddChild(&BangerDataManager);
+    asNode::AddChild(&BangerActiveManager);
+    asNode::AddChild(&BangerManager);
+    asNode::AddChild(&asnode34AF0);
+    asNode::AddChild(&PHYS);
+    asNode::AddChild(&RenderWeb);
+    asNode::AddChild(&Particles);
+
+    agiMeshCardVertex* vertex = NULL;
+
+    if (MMSTATE.Weather == mmWeather::Rain)
+        vertex = RainMeshCard;
+    else
+        vertex = agiMeshSet::DefaultQuad;
+
+    Particles.Init(100, 4, 4, 4, vertex);
+
+    SnowBirthRule.SetName("SnowRule");
+    AddChild(&SnowBirthRule);
+    SnowBirthRule.Load();
+
+    RainBirthRule.SetName("RainRule");
+    AddChild(&RainBirthRule);
+    RainBirthRule.Load();
+
+    BirthRule = 0;
+
+    Sky.Init("mmsky"_xconst);
+    Loader()->EndTask(0.0f);
+
+    BeginMemStat("mmCullCity.WEB");
+    RenderWeb.Load(name, 1);
+    EndMemStat();
+
+    HitIdBound = RenderWeb.HitIdBound;
+    HitIdBound->AddRef();
+
+    BuildingChain.Init(RenderWeb.MaxCells);
+    ObjectsChain.Init(RenderWeb.MaxCells);
+    ShadowChain.Init(RenderWeb.MaxCells);
+
+    InitObjectDetail();
+
+    LastInstance = (mmInstance*) mmInstanceHeap.HeapHead;
+    ResetInst = new mmYInstance();
+
+    LoadBangers(name);
+
+    if (MMSTATE.GameMode == mmGameMode::Blitz)
+    {
+        LoadBangers(arts_formatf<64>("%s_b%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Checkpoint)
+    {
+        LoadBangers(arts_formatf<64>("%s_r%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Circuit)
+    {
+        LoadBangers(arts_formatf<64>("%s_c%d", name, MMSTATE.EventId));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::Cruise)
+    {
+        LoadBangers(arts_formatf<64>("%s_roam", name));
+    }
+    else if (MMSTATE.GameMode == mmGameMode::CnR)
+    {
+        LoadBangers(arts_formatf<64>("%s_cops", name));
+    }
```

Comment:
Make this a switch instead of lots of if-elses

Reply:
I changed them to a switch, and also reordered them so they're consistent with the codebase.